### PR TITLE
fix(trino): handle all remaining common datatype parsing

### PR DESCRIPTION
**Pull request by cpcloud**
This PR fixes data type parsing issues coming from the trino backend, seen in https://github.com/ibis-project/ibis/discussions/6542.

### CODE REVIEW

File: ibis/backends/trino/tests/test_datatypes.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
+from __future__ import annotations
+
+import pytest
+from pytest import param
+
+import ibis.expr.datatypes as dt
+from ibis.backends.trino.datatypes import parse
+
+dtypes = [
+    # ("interval", dt.Interval(unit="s")),
```
Comment (jcrist) [inline ibis/backends/trino/tests/test_datatypes.py:10]: Do we want to leave this commented out?

**Review by jcrist — APPROVED** [review]
LGTM

File: ibis/backends/trino/tests/test_datatypes.py (hunk 1 above)
Comment (cpcloud) [reply ibis/backends/trino/tests/test_datatypes.py:10]: Whoops, thanks!

Comment (cpcloud) [reply ibis/backends/trino/tests/test_datatypes.py]: Fixed!

**Pull request merged**